compiler/rustc_middle/src/ty/sty.rs

```diff
@@ -1890,7 +1890,10 @@ impl<'tcx> TyS<'tcx> {
 
     #[inline]
     pub fn is_machine(&self) -> bool {
-        matches!(self.kind(), Int(..) | Uint(..) | Float(..))
+        // Yes, RawPtr is a "machine" type for these purposes.
+        // LLVM uses a vector-of-pointers model for scatter/gather ops,
+        // which typically use a base pointer and vector of signed integers.
+        matches!(self.kind(), Int(..) | Uint(..) | Float(..) | RawPtr(..))
     }
```

Comment:
This function looks generic and nothing in it says that its only use is the SIMD check.
I'd remove this function entirely and inline these variants into the match in `fn check_simd` instead.